iOSDFULibrary/Classes/Implementation/Firmware/DFUFirmware.swift

```
@@ -102,9 +101,8 @@ import Foundation
      
      - returns: The DFU firmware object or `nil` in case of an error.
      */
-    @objc convenience public init?(urlToZipFile: URL) {
-        self.init(urlToZipFile: urlToZipFile,
-                  type: DFUFirmwareType.softdeviceBootloaderApplication)
+    @objc convenience public init?(urlToZipFile: URL, errorHandler: ((String) -> ())?) {
```

The error handler should be `nil` by default, not only optional.

---

Also, the error handler should return the error, not a String.